src/snowflake/snowpark/modin/plugin/io/snow_io.py

```diff
@@ -177,6 +177,13 @@ def from_pandas(cls, df: pandas.DataFrame):
         """
         return cls.query_compiler_cls.from_pandas(df, pandas.DataFrame)
 
+    @classmethod
+    def json_normalize(cls, **kwargs):  # noqa: PR01
+        """
+        Normalize semi-structured JSON data into a query compiler representing a flat table.
+        """
+        return cls.from_pandas(pandas.json_normalize(**kwargs))
```

Comment:
should we mention in docs that we default to pandas for this? it's a bit of an edge case because the argument data is an in-memory object to begin with

Reply:
I don't think we should. As you said, the input data is already in-memory. So the fact that we're using native pandas internally is an implementation detail that the end user need not worry about.